# Scores and Selection Thresholds

Compare the mean and selected scores of five students. These are invented scores out of 100, not real student records. Only standard Python features are used.

## Inputs

Edit `scores` and `threshold`, then rerun through the calculation and output cells.

In [1]:
scores = [82, 75, 45, 90, 61]
threshold = 60

In [2]:
if not scores:
    raise ValueError("scores must not be empty")

mean_score = sum(scores) / len(scores)
selected = [score for score in scores if score >= threshold]

In [3]:
print("count:", len(scores))
print("mean:", mean_score)
print("threshold:", threshold)
print("selected:", selected)

count: 5
mean: 70.6
threshold: 60
selected: [82, 75, 90, 61]


## Interpretation

The default mean is 70.6. Scores at or above 60 are 82, 75, 90, and 61; 45 is excluded. The threshold controls selection only, so it does not change the overall mean.

## Change Conditions

1. A threshold of 80 selects `[82, 90]`.
2. Replacing the last score with 100 changes the mean to 78.4.
3. An empty list raises `ValueError`. Old output remaining after an error is not a new result.

Update the interpretation, restart the kernel, and run from top to bottom. Verify results and interpretation, then save.